src/CodeGenerator/FactoryGenerator.php

```diff
@@ -183,6 +183,8 @@ public function generate(string $class): string
         $factoryClassName                          = $this->namespace . '\\' . $this->buildClassName($class);
         [$namespace, $unqualifiedFactoryClassName] = $this->splitFullyQualifiedClassName($factoryClassName);
 
+        assert(is_string($this->outputDirectory));
```

Comment:
Imho, thats not safe. This is only true in case someone called the setter. Using php assert in this case is not how it should be used.

https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2021-08-02-TSC-Minutes.md#allow-php-assert-along-with-webmozart-assert-in-specific-cases

Reply:
Correct. Same as for the `AutoloadGenerator`.